boards/seeed/xiao_mg24/xiao_mg24.dts

```diff
@@ -0,0 +1,195 @@
+ */
+
+/dts-v1/;
+#include <silabs/efr32mg24b220f1536im48.dtsi>
+#include "xiao_mg24-pinctrl.dtsi"
+#include "seeed_xiao_connector.dtsi"
+
+/ {
+	model = "Seeed XIAO MG24";
+	compatible = "seeed,xiao_mg24", "silabs,efr32mg24";
+
+	chosen {
+		zephyr,console = &usart0;
+		zephyr,shell-uart = &usart0;
+		zephyr,uart-pipe = &usart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,bt-hci = &bt_hci_silabs;
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		led0 = &led0;
+		watchdog0 = &wdog0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			gpios = <&gpioa 7 GPIO_ACTIVE_LOW>;
+			label = "LED 0";
+		};
+	};
+};
+
+&cpu0 {
+	clock-frequency = <78000000>;
+};
+
+&pstate_em3 {
+	status = "disabled";
+};
+
+&hfxo {
+	status = "okay";
+	ctune = <95>;
+	precision = <50>;
+};
+
+&lfxo {
+	status = "okay";
+	ctune = <44>;
+	precision = <50>;
+};
+
+&hfrcodpll {
+	clock-frequency = <DT_FREQ_M(78)>;
+	clocks = <&hfxo>;
+	dpll-n = <3839>;
+	dpll-m = <1919>;
+	dpll-edge = "fall";
+	dpll-lock = "phase";
+	dpll-autorecover;
+};
+
+&em23grpaclk {
+	clocks = <&lfxo>;
+};
+
+&em4grpaclk {
+	clocks = <&lfxo>;
+};
+
+&sysrtcclk {
+	clocks = <&lfxo>;
+};
+
+&wdog0clk {
+	clocks = <&lfxo>;
+};
+
+&wdog1clk {
+	clocks = <&lfxo>;
+};
+
+&usart0 {
+	current-speed = <115200>;
+	pinctrl-0 = <&usart0_default>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&eusart0 {
+	compatible = "silabs,eusart-uart";
+	pinctrl-0 = <&eusart0_default>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+};
+
+&eusart1 {
+	compatible = "silabs,eusart-spi";
+	pinctrl-0 = <&eusart1_default>;
+	pinctrl-names = "default";
+	cs-gpios = <&gpioc 7 GPIO_ACTIVE_LOW>;
+	clock-frequency = <4000000>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+};
+
+&i2c0 {
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-names = "default";
+};
+
+&gpio {
+	location-swo = <0>;
+	status = "okay";
+};
+
+&gpioa {
+	status = "okay";
+};
+
+&gpiob {
+	status = "okay";
+};
+
+&gpioc {
+	status = "okay";
+
+};
+
+&gpiod {
+	status = "okay";
+};
+
+&wdog0 {
+	status = "okay";
+};
+
+&sysrtc0 {
+	status = "okay";
+};
+
+&se {
+	status = "okay";
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* Reserve 48 kB for the bootloader */
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x0 DT_SIZE_K(48)>;
+			read-only;
+		};
+
+		/* Reserve 720 kB for the application in slot 0 */
+		slot0_partition: partition@c000 {
+			label = "image-0";
+			reg = <0x0000c000 0x000B4000>;
+		};
+
+		/* Reserve 720 kB for the application in slot 1 */
+		slot1_partition: partition@C0000 {
+			label = "image-1";
+			reg = <0x000C0000 0x000B4000>;
+		};
+
+		/* Reserve 32 kB for the scratch partition */
+		scratch_partition: partition@174000 {
+			label = "image-scratch";
+			reg = <0x00174000 DT_SIZE_K(32)>;
+		};
+
```

Comment:
if sector sizes are the same then swap using scratch shouldn't be used in 2025

Reply:
Re-allocated those extra sectors into the two slots instead.